OpenRA.Mods.Common/Traits/Buildings/PrimaryBuilding.cs

```diff
@@ -71,8 +71,9 @@ Order IIssueOrder.IssueOrder(Actor self, IOrderTargeter order, Target target, bo
 
 		void IResolveOrder.ResolveOrder(Actor self, Order order)
 		{
-			if (order.OrderString == "PrimaryProducer")
-				SetPrimaryProducer(self, !IsPrimary);
+			var forceRallyPoint = order.OrderString == "SetRallyPoint" && order.ExtraData == 1u;
```

Comment:
Random numeric literals should be avoided. Either `RallyPoint.ForceSet` should be `public` and used here (`order.ExtraData == RallyPoint.ForceSet`), or better yet, a `public static bool IsForceSet(Order order)` in `RallyPoint` could be created:

Here:
```C#
			var forceRallyPoint = RallyPoint.IsForceSet(order);
```
In `RallyPoint`:
```C#
		public static bool IsForceSet(Order order)
		{
			return order.OrderString == "SetRallyPoint" && order.ExtraData == ForceSet;
		}
```